drivers/flash/flash_stm32_qspi.c

```diff
@@ -360,6 +361,49 @@ static bool qspi_address_is_valid(const struct device *dev, off_t addr,
 	return (addr >= 0) && ((uint64_t)addr + (uint64_t)size <= flash_size);
 }
 
+#ifdef CONFIG_STM32_XIP
+
+/* Function to return true if the quad-NOR flash is in MemoryMapped else false */
+static bool qspi_is_memorymapped(const struct device *dev)
```

Comment:
According to HAL team, we should be able to use the HAL_GetStatus API.

Reply:
> According to HAL team, we should be able to use the HAL_GetStatus API.

Update: This won't work as `HAL_GetStatus()` is based on driver internal variable (reset when switching from MCUBoot to Zephyr application), not hw setting